atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/bucketingthings/SweepAssignedBucketStoreKeyPersisterTest.java

```diff
@@ -0,0 +1,272 @@
+
+        assertThat(deserialisedSweepableBucket).isEqualTo(sweepableBucket);
+    }
+
+    @ParameterizedTest
+    @MethodSource("goldenSweepBucketCellsAndValues")
+    public void canDeserializeHistoricCellsAndValuesBackToSweepableBucket(
+            SweepableBucket sweepableBucket, Cell cell, Value value) {
+        SweepableBucket deserialisedSweepableBucket =
+                SweepAssignedBucketStoreKeyPersister.INSTANCE.fromSweepBucketCellAndValue(
+                        cell, value, TIMESTAMP_RANGE_PERSISTER);
+
+        assertThat(deserialisedSweepableBucket).isEqualTo(sweepableBucket);
+    }
+
+    @Test
+    public void sweepBucketAssignerStateMachineCellMatchesHistoricCell() {
+        Cell cell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketAssignerStateMachineCell();
+        assertThat(cell).isEqualTo(GOLDEN_SWEEP_BUCKET_ASSIGNER_STATE_MACHINE_CELL);
+    }
+
+    @Test
+    public void differentShardsAndStrategiesMapToDifferentSweepBucketPointerCells() {
+        Stream<ShardAndStrategy> shardsAndStrategies = Stream.of(
+                ShardAndStrategy.conservative(0),
+                ShardAndStrategy.conservative(1),
+                ShardAndStrategy.conservative(255),
+                ShardAndStrategy.thorough(0),
+                ShardAndStrategy.thorough(1),
+                ShardAndStrategy.thorough(212),
+                ShardAndStrategy.nonSweepable());
+
+        List<Cell> cells = shardsAndStrategies
+                .map(SweepAssignedBucketStoreKeyPersister.INSTANCE::sweepBucketPointerCell)
+                .collect(Collectors.toList());
+
+        assertThat(cells).doesNotHaveDuplicates();
+    }
+
+    @ParameterizedTest
+    @MethodSource("goldenSweepBucketPointerCells")
+    public void shardsAndStrategiesMapToHistoricSweepBucketPointerCells(
+            ShardAndStrategy shardAndStrategy, Cell expectedCell) {
+        Cell actualCell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketPointerCell(shardAndStrategy);
+        assertThat(actualCell).isEqualTo(expectedCell);
+    }
+
+    @Test
+    public void differentBucketIdentifiersMapToDifferentSweepBucketRecordsCells() {
+        Set<Integer> bucketIdentifiers = Set.of(0, 1, 2, 123123, Integer.MAX_VALUE);
+        List<Cell> cells = bucketIdentifiers.stream()
+                .map(SweepAssignedBucketStoreKeyPersister.INSTANCE::sweepBucketRecordsCell)
+                .collect(Collectors.toList());
+        assertThat(cells).doesNotHaveDuplicates();
+    }
+
+    @ParameterizedTest
+    @MethodSource("goldenSweepBucketRecordsCells")
+    public void bucketIdentifiersMapToHistoricSweepBucketRecordsCells(long bucketIdentifier, Cell expectedCell) {
+        Cell actualCell = SweepAssignedBucketStoreKeyPersister.INSTANCE.sweepBucketRecordsCell(bucketIdentifier);
+        assertThat(actualCell).isEqualTo(expectedCell);
+    }
+
+    // TODO(mdaudali): Should we add validation to ensure that bucket identifier for bucket and passed in above >= 0?
```

Comment:
Open q.

Reply:
Yeah this seems reasonable to add (doesn't need to be in this one if inconvenient)